compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirMemberPropertyChecker.kt

```diff
@@ -71,14 +71,14 @@ object FirMemberPropertyChecker : FirRegularClassChecker() {
                 reporter.reportOn(it, FirErrors.ABSTRACT_DELEGATED_PROPERTY, context)
             }
 
-            checkAccessor(property.getter, property.delegate) { src, _ ->
-                reporter.reportOn(src, FirErrors.ABSTRACT_PROPERTY_WITH_GETTER, context)
+            checkAccessor(property.getter, property.delegate) { src, _, hasBody ->
+                if (hasBody) reporter.reportOn(src, FirErrors.ABSTRACT_PROPERTY_WITH_GETTER, context)
             }
-            checkAccessor(property.setter, property.delegate) { src, symbol ->
+            checkAccessor(property.setter, property.delegate) { src, symbol, hasBody ->
                 if (symbol.fir.visibility == Visibilities.Private && property.visibility != Visibilities.Private) {
                     reporter.reportOn(src, FirErrors.PRIVATE_SETTER_FOR_ABSTRACT_PROPERTY, context)
                 } else {
-                    reporter.reportOn(src, FirErrors.ABSTRACT_PROPERTY_WITH_SETTER, context)
+                    if (hasBody) reporter.reportOn(src, FirErrors.ABSTRACT_PROPERTY_WITH_SETTER, context)
```

Comment:
This could be `else if (hasBody)`. Then, I guess IJ will suggest to refactor this to `when`?